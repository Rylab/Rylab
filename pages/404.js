import Head from 'next/head'
import Image from 'next/image'
import Link from 'next/link'
import { BASE_URL, SITE_TITLE } from '../utils/constants'

const pageTitle = `${SITE_TITLE} :: Not Found`

export default function NotFound() {
  return (
    <>
      <Head>
        <link rel="canonical" href={`${ BASE_URL }/404`} />
        <title>{ pageTitle }</title>
      </Head>
      <div className="content" style={{ marginTop: 100 }}>
        <Link href={'/'} className="discreet" passHref>
          <Image
            className="crosshair"
            height="300"
            width="300"
            src="/img/bsd_extrovert.png" alt="RyLaB Home" />
        </Link>
        <p style={{ color: '#666', marginBottom: 250, marginTop: 20 }}>
          <b>&lt;404&gt;</b>&nbsp;Not Found&nbsp;<b>&lt;/404&gt;</b>
        </p>
      </div>
    </>
  )
}
