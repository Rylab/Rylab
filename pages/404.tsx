import Head from 'next/head'
import Image from 'next/image'
import Link from 'next/link'
import Layout from '../components/Layout'
import { BASE_URL, SITE_TITLE } from '../utils/constants'

const pageTitle = `${SITE_TITLE} :: Not Found`

const NotFound = (): JSX.Element => {
  return (
    <>
      <Head>
        <link rel="canonical" href={`${BASE_URL}/404`} />
        <title>{pageTitle}</title>
      </Head>
      <div className="content" style={{ marginTop: 100 }}>
        <Link href={'/'} className="discreet" passHref>
          <Image
            className="crosshair"
            height="300"
            width="300"
            src="/img/bsd_extrovert.webp" alt="RyLaB Home" />
        </Link>
        <p style={{ color: '#666', marginBottom: 250, marginTop: 20 }}>
          <b>&lt;404&gt;</b>&nbsp;Not Found&nbsp;<b>&lt;/404&gt;</b>
        </p>
      </div>
    </>
  )
}

NotFound.getLayout = page => <Layout hideAdminInput>{page}</Layout>

export default NotFound
