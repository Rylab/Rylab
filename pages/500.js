import Head from 'next/head'
import Image from 'next/image'
import Link from 'next/link'
import { baseUrl, siteTitle } from '../components/layout'

const pageTitle = `${siteTitle} :: Internal Server Error`

export default function InternalServerError() {
  return (
    <>
      <Head>
        <link rel="canonical" href={`https://${ baseUrl }/500`} />
        <title>{ pageTitle }</title>
      </Head>
      <div className="content" style={{ marginTop: 100 }}>
        <Link href={'/'} className="discreet" passHref>
          <Image
            className="crosshair"
            height="300"
            width="300"
            src="/img/bsd_extrovert.png" alt="RyLaB Home" />
        </Link>
        <p style={{ color: '#666', marginBottom: 250, marginTop: 20 }}>
          <b>&lt;500&gt;</b>&nbsp;Internal Server Error&nbsp;<b>&lt;/500&gt;</b>
        </p>
      </div>
    </>
  )
}
