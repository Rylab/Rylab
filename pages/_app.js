import { createContext, useEffect, useState } from 'react'
import Head from 'next/head'
import { useRouter } from 'next/router'

import '../styles/global.css'
import Layout from '../components/Layout'
import { initUuid } from '../utils/helpers'

export const AppContext = createContext()
export const jsonType = 'application/json'

export const getHeaders = ({ uuid, password }) => {
  const headers = {
    accept: jsonType,
    'content-type': jsonType,
    'x-uuid': uuid ?? 'anonymous',
  }

  if (password) headers['x-admin'] = password

  return headers
}

export default function App({ Component, pageProps }) {
  const router = useRouter()
  const [password, setPassword] = useState('')
  const [uuid, setUuid] = useState('')

  const getLayout = Component.getLayout || (page => <Layout>{page}</Layout>)

  useEffect(() => {
    if (router.isReady) {
      if (!uuid) {
        const uuid = initUuid()
        setUuid(uuid)
      }

      const localPass = localStorage.getItem('managePass')

      if (localPass) {
        const bufferPass = Buffer.from(localPass, 'base64')
        const managePass = bufferPass.toString('utf8')

        setPassword(managePass ?? '')
      }
    }
  }, [router]) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <AppContext.Provider value={{
      password,
      uuid,
      setPassword,
      setUuid,
    }}>
        <Head>
          <meta name="viewport" content="width=device-width, initial-scale=1" />
        </Head>
      { getLayout(<Component {...pageProps} />) }
    </AppContext.Provider>
  )
}
