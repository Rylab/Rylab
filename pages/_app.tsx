import Head from 'next/head'
import { useRouter } from 'next/router'
import { createContext, JSX, useEffect, useState } from 'react'

import '../styles/global.css'
import Layout from '../components/Layout'
import { initUuid } from '../utils/helpers'

interface ContextProps {
  password: string
  setPassword: (password: string) => void
  showLogin: boolean
  setShowLogin: (show: boolean) => void
  uuid: string
  setUuid: (uuid: string) => void
}

export const AppContext = createContext({
  password: '',
  setPassword: (password: string) => {},
  showLogin: false,
  setShowLogin: (show: boolean) => {},
  uuid: '',
  setUuid: (uuid: string) => {},
})

export const JsonType = 'application/json'

interface HeaderProps {
  uuid: string
  password: string
}

export const getHeaders = ({ uuid, password }: HeaderProps) => {
  const headers = {
    'accept': JsonType,
    'content-type': JsonType,
    'x-uuid': uuid ?? 'anonymous',
    'x-admin': '',
  }

  if (password) headers['x-admin'] = password

  return headers
}

interface AppProps {
  Component: any
  pageProps: any
}

export default function App({ Component, pageProps }: AppProps) {
  const [password, setPassword] = useState('')
  const router = useRouter()
  const [showLogin, setShowLogin] = useState(false)
  const [uuid, setUuid] = useState('')

  const getLayout = Component.getLayout || ((page: JSX.Element) => <Layout>{page}</Layout>)

  useEffect(() => {
    if (router.isReady) {
      if (!uuid) {
        const uuid = initUuid()
        setUuid(uuid)
      }

      const localPass = localStorage.getItem('managePass')

      if (localPass) {
        const bufferPass = Buffer.from(localPass, 'base64')
        const managePass = bufferPass.toString('utf8')

        setPassword(managePass ?? '')
      }
    }
  }, [router]) // eslint-disable-line react-hooks/exhaustive-deps

  const contextValue: ContextProps = {
    password, setPassword,
    showLogin, setShowLogin,
    uuid, setUuid,
  }

  return (
    <AppContext.Provider value={contextValue}>
      <Head>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
      </Head>
      {getLayout(<Component {...pageProps} />)}
    </AppContext.Provider>
  )
}
