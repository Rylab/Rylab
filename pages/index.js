import Head from 'next/head'
import Link from 'next/link'
import { useEffect, useState } from 'react'
import { siteTitle } from '../components/layout'
import LoadingSpinner from '../components/LoadingSpinner'

const contentType = 'application/json'

export default function Index() {
  const [loading, setLoading] = useState(true)
  const [selves, setSelves] = useState({})

  useEffect(() => {
    getSelves()
  }, [])

  const getSelves = async () => {
    try {
      setLoading(true)
  
      const res = await fetch('/api/selves', {
        headers: {
          accept: contentType,
          'content-type': contentType,
        },
        method: 'GET',
      })
      const selves = await res.json()
      setLoading(false)
  
      if (selves.data) {
        setSelves({ data: selves.data })
      } else {
        console.error(res)
      }
    } catch (error) {
      setLoading(false)
      console.error(error)
    }
  }

  return (
    <>
      <Head>
        <link rel="canonical" href="https://rylab.com" />
        <title>{ siteTitle } :: Home</title>
        <meta name="og:title" content={`${siteTitle} :: Home`} />
        <meta name="description" content="Welcome to Rylab, digital home of Ryan LaBarre" />
        <meta property="og:description" content="Digital Home of Ryan LaBarre" />
      </Head>
      <main>
        <div id="header" className="hoverdots">
          <img className="rylab"
            src="/img/rylab_extrovert.png"
            alt="A very pretty building in San Francisco. &copy; Ryan D LaBarre"
            title="Pretty building in San Francisco. Rylab does not live here (but has been here)" />
        </div>
        <div id="rylab">(: hello :)</div>
        <div id="content">
          <p id="haiku-today" className="large light">
            someday&nbsp;I will&nbsp;make,<br />
            a&nbsp;great personal&nbsp;website;<br />
            it&nbsp;may be&nbsp;today
          </p>
          <p id="haiku-tomorrow" className="small dark hoverlight">
            perhaps tomorrow?<br />
            probably&nbsp;not then&nbsp;either...<br />
            our&nbsp;planet still&nbsp;spins<br />
          </p>
          <LoadingSpinner />
          <div id="digitalhaiku">
            <a href="https://medium.com/@rylab/abc-digitally-evolving-futurist-goals-9b11bd0f54d2"
              rel="noreferrer" target="_medium" title="My Haiku on Medium">
              #digitalhaiku
            </a>
          </div>
          <div id="demos">
            <Link href="/demos/tapespinner" title="Tape Spinner demo">
              SVG Cassette Tape Spinner
            </Link>
          </div>
          <div id="saxylab">
            <a href="https://saxylab.com" target="_saxylab" title="Saxy and Ryan Wedding :: SaxyLab">
              Saxy and Ryan got hitched!
            </a>
          </div>
          <br />
          {selves.data && selves.data.length ? (
            <>
              <h1 style={{ marginBottom: 20 }}>Digital Selves</h1>
              <ul className="narrow list">
                {selves.data.map(link => (
                  <li className={`link ${link.alias}`} key={link.alias}>
                    <a title={link.title} href={link.url} rel="noreferrer" target="_blank">
                      {link.name}</a>
                  </li>
                ))}
              </ul>
            </>
          ) : (
            <LoadingSpinner />
          )}
        </div>
        <a id="aeq" href="https://algorithmeq.com" target="_aeq">
          <img className="bsd" src="/img/bsd_extrovert.png" alt="// MacOS <== ++BSD;" /></a>
      </main>
    </>
  )
}
