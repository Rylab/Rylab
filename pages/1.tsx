import dynamic from 'next/dynamic'
import Head from 'next/head'
import { useRouter } from 'next/router'
import { useContext, useEffect, useState } from 'react'

import { AppContext, getHeaders } from './_app'
import { Navigation, TapeAdder, TapeSpinner, tapeColors } from '../components'
import { BASE_URL, MAX_LINE_LENGTH, SITE_TITLE } from '../utils/constants'
import { getSongEmbed, validateUuid } from '../utils/helpers'

const pageTitle = `${SITE_TITLE} :: Your Tape Collections`

const HankoProfile = dynamic(() => import('../components/HankoProfile'), { ssr: false })

export default function DashboardForUuid() {
  const [canAdd, setCanAdd] = useState(false)
  const { password, uuid } = useContext(AppContext)
  const [songs, setSongs] = useState([])
  const [pageUuid, setPageUuid] = useState('')
  const [loading, setLoading] = useState(true)

  const router = useRouter()

  // wait until router is ready and client uuid is set to fetch songs
  useEffect(() => {
    if (router.isReady && router.query.uuid && uuid) {
      const decodedUuid = validateUuid(decodeURI(router.query.uuid.toString()))

      if (decodedUuid) {
        setPageUuid(decodedUuid)
        setCanAdd(decodedUuid === uuid)
        getSongs(decodedUuid)
      }
    }
  }, [router, password, uuid]) // eslint-disable-line react-hooks/exhaustive-deps

  // fetch songs by uuid
  const getSongs = async targetUuid => {
    try {
      setLoading(true)

      const safeUrl = `/api/songs/${validateUuid(targetUuid)}`
      const songRes = await fetch(safeUrl, {
        headers: getHeaders({ uuid, password }),
        method: 'GET',
      }).then(res => res.json())

      if (songRes?.data) {
        songRes.data.map(song => {
          if (typeof tapeColors !== 'undefined' && tapeColors.length && !song.style)
            song.style = { backgroundColor: tapeColors[Math.floor(Math.random() * tapeColors.length)] }
        })

        setSongs(songRes.data)
      } else {
        console.error(songRes)
      }
    } catch (error) {
      console.error(error)
    } finally {
      setLoading(false)
    }
  }

  return (
    <>
      <Head>
        <link rel="canonical" href={`${BASE_URL}/1`} />
        <title>{pageTitle}</title>
        <meta name="og:title" content={pageTitle} />
        <meta name="description" content="Your generated and curated AI powered tape collections." />
        <meta property="og:description" content="Your generated and curated AI powered tape collections." />
      </Head>
      <main>
        <Navigation path={'1'} />
        {!loading && songs?.map(song => {
          const hasLongArtist = song.artist.length > MAX_LINE_LENGTH
          const hasLongTitle = song.title.length > MAX_LINE_LENGTH

          return (
            <TapeSpinner style={song.style} spin={song.spin} key={song._id} id={`#${song._id}`}>
              <div title={song.title} onClick={() => getSongEmbed(song._id)} className={`titleLine${hasLongTitle ? ' long' : ''}`}>
                {song.title}</div>
              <div title={song.artist} onClick={() => getSongEmbed(song._id)} className={`artistLine${hasLongArtist ? ' long' : ''}`}>
                {song.artist}</div>
              <div className="disabled uuidLine">{song.uuid}</div>
              <div className="songIdLine" onClick={() => getSongEmbed(song._id)}>{song._id}</div>
            </TapeSpinner>
          )
        })}
        {canAdd && <TapeAdder />}
        <div className='flex flex-center mt-60'><HankoProfile /></div>
      </main>
    </>
  )
}
