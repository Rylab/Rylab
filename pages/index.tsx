import Head from 'next/head'
import Link from 'next/link'
import { useContext, useEffect, useState } from 'react'
import languages from '../constants/languages'
import { AppContext, getHeaders } from './_app'
import LoadingSpinner from '../components/LoadingSpinner'
import { BASE_URL, SITE_TITLE } from '../utils/constants'

const pageTitle = `${SITE_TITLE} :: Home`

export default function Index() {
  const { password, uuid } = useContext(AppContext)
  const [loading, setLoading] = useState(true)
  const [selves, setSelves] = useState({
    data: [],
  })

  const getSelves = async () => {
    if (!uuid) return

    try {
      setLoading(true)

      const selves = await fetch('/api/selves', {
        headers: getHeaders({ uuid, password }),
        method: 'GET',
      }).then(res => res.json())

      if (selves.data) {
        setSelves({ data: selves.data })
      } else {
        console.error(selves)
      }
    } catch (error) {
      console.error(error)
    } finally {
      setLoading(false)
    }
  }

  useEffect(() => {
    getSelves()
  }, [password, uuid]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    console.table(languages.map(lang => { return { 'Language': lang } }))
  }, [])

  return (
    <>
      <Head>
        <link rel="canonical" href={BASE_URL} />
        <link rel="preload" href="/img/bsd_extrovert.webp" as="image" fetchPriority="high" type="image/webp" />
        <link rel="preload" href="/img/bsd_introvert.webp" as="image" fetchPriority="high" type="image/webp" />
        <title>{pageTitle}</title>
        <meta name="og:title" content={pageTitle} />
        <meta name="description" content="Welcome to Rylab, digital home of Ryan LaBarre" />
        <meta property="og:description" content="Digital Home of Ryan LaBarre" />
      </Head>
      <main>
        <div className="header hoverborder lockish">
        </div>
        <div className="rylab-copy lockish">(: hello :)</div>
        <div className="content lockish">
          <p className="haiku-today large light hoverdark">
            someday&nbsp;I will&nbsp;make,<br />
            a&nbsp;great personal&nbsp;website;<br />
            it&nbsp;may be&nbsp;today
          </p>
          <p className="haiku-tomorrow small dark hoverlight">
            perhaps tomorrow?<br />
            probably&nbsp;not then&nbsp;either<br />
            our&nbsp;planet still&nbsp;spins<br />
          </p>
          <div className="narrow list">
            <div className="demo">
              <Link href="/jamband" title="Jam band generator | matchmaker">
                Jam Band Generator
              </Link>
            </div>
            <div className="demo">
              <Link href="/demos/tapespinner" title="Tape Spinner demo | playground">
                Tape Spinner demo
              </Link>
            </div>
            <div className="digitalhaiku">
              <a href="https://medium.com/@rylab/abc-digitally-evolving-futurist-goals-9b11bd0f54d2"
                rel="noreferrer" target="_medium" title="My Haiku on Medium">
                #digitalhaiku
              </a>
            </div>
            <div className="saxylab">
              <a href="https://saxylab.com" target="_saxylab" title="Saxy and Ryan :: SaxyLab">
                SaxyLab
              </a>
            </div>
          </div>
          <br />
          <h1 style={{ marginBottom: 20, marginTop: 40 }}>Digital Selves</h1>
          {selves.data?.length ? (
            <ul className="narrow list">
              {selves.data.map(link => (
                <li className={`link ${link.alias}`} key={link.alias}>
                  <a className="link" title={link.title} href={link.url} rel="noreferrer" target="_blank">
                    {link.name}</a>
                </li>
              ))}
            </ul>
          ) : (
            loading ? <LoadingSpinner /> : <span className="dark">[]</span>
          )}
        </div>
        <img className="bsd lockish" src="/img/bsd_extrovert.webp" alt="// MacOS <== ++BSD;" />
      </main>
    </>
  )
}
