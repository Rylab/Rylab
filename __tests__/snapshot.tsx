import { render } from '@testing-library/react'
import { AppContext } from '../pages/_app'
import Index from '../pages/index'

it('renders homepage unchanged', () => {
  const { container } = render(
    <AppContext.Provider value={{
      password: '',
      uuid: '',
      setPassword: jest.fn(),
      setUuid: jest.fn(),
    }}>
      <Index />
    </AppContext.Provider>
  )

  expect(container).toMatchSnapshot()
})
