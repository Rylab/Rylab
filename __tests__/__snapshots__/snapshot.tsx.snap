// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main>
    <div
      class="header hoverborder lockish"
    />
    <div
      class="rylab-copy lockish"
    >
      (: hello :)
    </div>
    <div
      class="content lockish"
    >
      <p
        class="haiku-today large light hoverdark"
      >
        someday I will make,
        <br />
        a great personal website;
        <br />
        it may be today
      </p>
      <p
        class="haiku-tomorrow small dark hoverlight"
      >
        perhaps tomorrow?
        <br />
        probably not then either
        <br />
        our planet still spins
        <br />
      </p>
      <div
        class="narrow list"
      >
        <div
          class="demo"
        >
          <a
            href="/demos/tapeai"
            title="Tape AI playground"
          >
            Tape AI (music cassette maker)
          </a>
        </div>
        <div
          class="demo"
        >
          <a
            href="/demos/tapespinner"
            title="Tape Spinner playground"
          >
            Animated SVG cassettes
          </a>
        </div>
        <div
          class="digitalhaiku"
        >
          <a
            href="https://medium.com/@rylab/abc-digitally-evolving-futurist-goals-9b11bd0f54d2"
            rel="noreferrer"
            target="_medium"
            title="My Haiku on Medium"
          >
            #digitalhaiku
          </a>
        </div>
        <div
          class="saxylab"
        >
          <a
            href="https://saxylab.com"
            target="_saxylab"
            title="Saxy and Ryan :: SaxyLab"
          >
            SaxyLab
          </a>
        </div>
        <div
          class="sc-aXZVg iXnuBG"
          data-testid="loading-spinner"
        >
          <svg
            class="spinner"
            viewBox="0 0 50 50"
          >
            <lineargradient
              gradientTransform="rotate(120)"
              id="linear"
              x1="0%"
              x2="100%"
              y1="0%"
              y2="0%"
            >
              <stop
                offset="5%"
                stop-color="#010101"
              />
              <stop
                offset="95%"
                stop-color="#343434"
              />
            </lineargradient>
            <circle
              class="path"
              cx="25"
              cy="25"
              fill="url(#linear)"
              r="20"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
      <br />
      <h1
        style="margin-bottom: 20px; margin-top: 40px;"
      >
        Digital Selves
      </h1>
      <div
        class="sc-aXZVg iXnuBG"
        data-testid="loading-spinner"
      >
        <svg
          class="spinner"
          viewBox="0 0 50 50"
        >
          <lineargradient
            gradientTransform="rotate(120)"
            id="linear"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              offset="5%"
              stop-color="#010101"
            />
            <stop
              offset="95%"
              stop-color="#343434"
            />
          </lineargradient>
          <circle
            class="path"
            cx="25"
            cy="25"
            fill="url(#linear)"
            r="20"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <img
      alt="// MacOS <== ++BSD;"
      class="bsd lockish"
      src="/img/bsd_extrovert.webp"
    />
  </main>
</div>
`;
